packages/terra-search-field/src/SearchField.jsx

```diff
@@ -0,0 +1,94 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Button from 'terra-button';
+import IconSearch from 'terra-icon/lib/icon/IconSearch';
+import classNames from 'classnames';
+import './SearchField.scss';
+
+const propTypes = {
+  /**
+   * Placeholder text to show while the search field is empty.
+   */
+  placeholder: PropTypes.string,
+
+  /**
+   * The minimum number of characters to perform a search.
+   */
+  minimumSearchTextLength: PropTypes.number,
+
+  /**
+   * How long the component should wait after input before performing an automatic search.
+   */
+  searchDelay: PropTypes.number,
+
+  /**
+   * A callback to perform search. Sends parameter {String} searchText.
+   */
+  onSearch: PropTypes.func,
+};
```

Comment:
I'd like to get some other thoughts on this before making the change.  Is there a concern about exposing too much about the implementation through props?  Would it be better to selectively expose functionality as needed?

I can see the value in making this change,  but I worry about locking in the implementation if props are exposed for each specific child component.